src/main/java/HintGenerator/HintGenerator.java

```diff
@@ -0,0 +1,72 @@
+package HintGenerator;
+
+import entities.DesignableMaze;
+import entities.Maze;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.HashSet;
+import java.util.Set;
+
+public class HintGenerator {
+
+    private static List<int[]> getValidMoves(DesignableMaze maze, int position[]) {
```

Comment:
refer to comment 

https://github.com/CSC207-2022F-UofT/course-project-wasted-potential/pull/17/files#r1028530528